### ISSUE TITLE
category based based presentation of iwdescr's keywords/properties

### ISSUE THREAD

**Issue by nbehrnd** [IanAWatson/Molecular_Property_Profile#1]
To ease to get familiar with the keywords in [iwdescr](https://github.com/IanAWatson/LillyMol-4.0-Bazel/blob/master/src/Molecule_Tools/iwdescr.cc)'s scrutiny and what properties they represent, I reused a concept of reorganization applied a few days before [here](https://github.com/Beliavsky/Fortran-related-books/pull/5).

The sort provided by script `reorganizer.sh` only considers the 44 entries in file [column_descriptions.txt](https://github.com/IanAWatson/Molecular_Property_Profile/blob/main/column_descriptions.txt) as available by today because it wasn't obvious for me how this (as a source file) was generated in first place.  In the longer run, a more compact presentation (file `properties.txt` only serves as an example) could cover all 200 properties.

[column_descriptions.txt](https://github.com/IanAWatson/Molecular_Property_Profile/files/9027702/column_descriptions.txt)
[properties.txt](https://github.com/IanAWatson/Molecular_Property_Profile/files/9027703/properties.txt)
[reorganizer.sh.txt](https://github.com/IanAWatson/Molecular_Property_Profile/files/9027704/reorganizer.sh.txt)

**Comment by IanAWatson** [comment]
This looks OK, maybe a little more complex that it needs to be, but fine.

The format of column_descriptions is a protocol buffer, and so there are parsers for that in various languages. This particular version of the file was manually generated as a one-time thing.

In terms of the other features from iwdescr, some are likely of interest in comparing collections, some less so. For this first release, I focused on what seem to be the most commonly used, and which we have found to be useful comparing collections.

Since the first release, I have encountered problems with collections that are outside the bounds of the reference sets included - fragments or larger molecules. I hope to have some fixes to better handle those cases, but it gets messy.

I also hope to have more in-detail descriptions of the features iwdescr generates. The format for what that should be is undecided so far, but could quite reasonably be as you have here.

Do you want to make a pull request for this? I don't think I can easily merge this? Also, the .sh script should not end in .txt.

Thanks!

Ian

**Comment by IanAWatson** [comment]
Sorry, did not mean to mark this as closed,

**Comment by nbehrnd** [comment]
I fully agree that the approach taken to reformat the original pattern into a .csv _like_ pattern (first part of the transformation) is/was a bit long, even more so if there is a parser to provide the same right from the beginning.  The script's inclusion _in the discussion_ only aimed to document how I derived the more compact representation now deposit in a separate PR [here](https://github.com/IanAWatson/Molecular_Property_Profile/pull/2).  Since not opting for a code block, appending the .txt file extension to the .sh for the bash script only reflected the constraints set by GitHub for uploads.  Which mainly are either  pure text (.txt, .csv, .log, .md), binary illustrations like .png, and archive .zip/.gz; but -- as I understand -- not something potentially executable.
